src/components/Settings/integrations/Organization/OrganizationAccordion.tsx

```diff
@@ -83,16 +87,19 @@ export const OrganizationAccordion: React.FC<OrganizationAccordionProps> = ({
   handleAccordionChange,
   expandedPanel,
 }) => {
-  const { t } = useTranslation();
-  const accountListId = useAccountListId();
-  const { enqueueSnackbar } = useSnackbar();
-  const { appName } = useGetAppSettings();
+  const [selectedOrganization, setSelectedOrganization] =
```

Comment:
Instead of having one `useState` for the organization and another `useState` for whether the modal is open, I would remove the `selectedOrganization` use state and do something like this for the modals connected to an organization:

```tsx
const [importDataSyncModalOrg, setImportDataSyncModalOrg] = <OrganizationAccountPartial | null>(null);

// Open the modal
importDataSyncModalOrg(org);

// Render the modal
{addAccountModalOrg && <OrganizationImportDataSyncModal>}
```

Reply:
Great idea!